hat/assets/js/apps/Iaso/domains/forms/components/Filters.tsx

```diff
@@ -1,53 +1,60 @@
 import React, { useState, FunctionComponent, useCallback } from 'react';
 
+import Add from '@mui/icons-material/Add';
+import SearchIcon from '@mui/icons-material/Search';
 import { Grid, Button, Box, useMediaQuery, useTheme } from '@mui/material';
 import { makeStyles } from '@mui/styles';
-import SearchIcon from '@mui/icons-material/Search';
-import { commonStyles, useSafeIntl } from 'bluesquare-components';
+import {
+    commonStyles,
+    useRedirectTo,
+    useSafeIntl,
+} from 'bluesquare-components';
 
+import { baseUrls } from 'Iaso/constants/urls';
+import * as Permission from 'Iaso/utils/permissions';
+import { DisplayIfUserHasPerm } from '../../../components/DisplayIfUserHasPerm';
 import InputComponent from '../../../components/forms/InputComponent';
 import { useFilterState } from '../../../hooks/useFilterState';
 
-import MESSAGES from '../messages';
-
-import { baseUrl } from '../config';
+import { useGetOrgUnitTypesDropdownOptions } from '../../orgUnits/orgUnitTypes/hooks/useGetOrgUnitTypesDropdownOptions';
 import { useGetPlanningsOptions } from '../../plannings/hooks/requests/useGetPlannings';
 import { useGetProjectsDropdownOptions } from '../../projects/hooks/requests';
-import { useGetOrgUnitTypesDropdownOptions } from '../../orgUnits/orgUnitTypes/hooks/useGetOrgUnitTypesDropdownOptions';
+import { baseUrl } from '../config';
+import { Params } from '../index';
+import MESSAGES from '../messages';
 
 const useStyles = makeStyles(theme => ({
     ...commonStyles(theme),
 }));
 
-type Params = {
-    pageSize: string;
-    order: string;
-    page: string;
-    search?: string;
-    showDeleted?: string;
-    planning?: string;
-    projectsIds?: string;
-};
-
 type Props = {
     params: Params;
 };
 
 const Filters: FunctionComponent<Props> = ({ params }) => {
     const classes: Record<string, string> = useStyles();
     const { formatMessage } = useSafeIntl();
+    const redirectTo = useRedirectTo();
     const { filters, handleSearch, handleChange, filtersUpdated } =
-        useFilterState({ baseUrl, params, withPagination: false });
+        useFilterState({
+            baseUrl,
+            params,
+            withPagination: false,
+            searchActive: 'isSearchActive',
+            searchAlwaysEnabled: true,
+        });
     const [textSearchError, setTextSearchError] = useState<boolean>(false);
-    const [showDeleted, setShowDeleted] = useState<boolean>(
-        filters.showDeleted === 'true',
-    );
     const handleShowDeleted = useCallback(
         (key, value) => {
-            // converting false to undefined to be able to compute `filtersUpdated` correctly
-            const valueForParam = value || undefined;
+            const valueForParam = value ? 'true' : undefined;
+            handleChange(key, valueForParam);
+        },
+        [handleChange],
+    );
+    const handleShowInstancesCount = useCallback(
```

Comment:
There's a `useCheckboxState` hook in useFilterState.ts that could be used (FYI), but I'm not sure the code would gain much clarity in this particular case